### ISSUE TITLE
Testing for Reselling with Royalties Fails 

### ISSUE THREAD

**Issue by idea404** [near-examples/nft-tutorial#28]
Currently there is no testing flow for reselling an NFT with royalties. When implementing this with `near-workspaces-ava`, the asserts fail when running the following simulation test:

```typescript
workspace.test(
  "cross contract: reselling and royalties",
  async (test, { nft_contract, market_contract, alice, bob, charlie }) => {
    const royalties_string = `{"${alice.accountId}":2000}`;
    const royalties = JSON.parse(royalties_string);
    test.log("royalties: ", royalties);
    await mintNFT(alice, nft_contract, royalties);
    await payForStorage(alice, market_contract);
    const ask_price = "300000000000000000000000";
    await placeNFTForSale(market_contract, alice, nft_contract, ask_price);

    // offer for higher price
    const alice_balance_before_offer = (await alice.balance()).total.toBigInt();
    const bid_price = ask_price + "0";
    await purchaseListedNFT(nft_contract, bob, market_contract, bid_price);
    const alice_balance_after_offer = (await alice.balance()).total.toBigInt();
    const alice_balance_difference = (
      alice_balance_after_offer - alice_balance_before_offer
    ).toString();

    // assert alice gets paid
    test.is(
      alice_balance_difference.substring(0, 3),
      bid_price.substring(0, 3),
      "Expected alice balance to roughly increase by sale price"
    );

    // bob relists NFT for higher price
    test.log("bob paying for storage");
    await payForStorage(bob, market_contract);
    const resell_ask_price = bid_price + "0";
    test.log("bob placing NFT for sale");
    await placeNFTForSale(market_contract, bob, nft_contract, resell_ask_price);
  
    // bob updates price to lower ask
    test.log("bob updating NFT price for lower");
    const lowered_resell_ask_price = "600000000000000000000000";
    const update_price_payload = {
      nft_contract_id: nft_contract,
      token_id: "TEST123",
      price: lowered_resell_ask_price,
    };
    await bob.call(
      market_contract,
      "update_price",
      update_price_payload,
      defaultCallOptions(DEFAULT_GAS, "1")
    );

    // charlie buys NFT from bob
    test.log("charlie purchasing bobs NFT");
    const alice_balance_before_offer_2 = (
      await alice.balance()
    ).total.toBigInt();
    const bob_balance_before_offer = (await bob.balance()).total.toBigInt();
    test.log("bob_balance_before_offer", bob_balance_before_offer);
    await purchaseListedNFT(
      nft_contract,
      charlie,
      market_contract,
      resell_ask_price
    );
    const alice_balance_after_offer_2 = (
      await alice.balance()
    ).total.toBigInt();
    const alice_balance_difference_2 = (
      alice_balance_after_offer_2 - alice_balance_before_offer_2
    ).toString();
    const bob_balance_after_offer = (await bob.balance()).total.toBigInt();
    test.log("bob_balance_after_offer", bob_balance_after_offer);
    const bob_balance_difference = (
      bob_balance_after_offer - bob_balance_before_offer
    ).toString();

    // assert alice gets paid royalties
    // TODO: this fails on sandbox and should pass 
    test.is(
      alice_balance_difference_2.substring(0, 2),
      "120", // 20% of lowered_resell_ask_price
      "Expected bob balance to roughly increase by 80% of sale price"
    );
    // assert bob gets paid
    test.is(
      bob_balance_difference.substring(0, 2),
      "480", // 80% of lowered_resell_ask_price
      "Expected bob balance to roughly increase by 80% of sale price"
    );
  }
);
```

**Comment by njelich** [comment]
You didnt await the purchaseListedNFT, when that is awaited the outputs change, though they are still wrong

**Comment by idea404** [comment]
Good point @njelich, was very helpful and put me in the right path to fix the test! Have now included it in #22
